dist/datasource.ts

```diff
@@ -205,58 +205,60 @@ export default class IrondbDatasource {
       headers['X-Circonus-Account'] = this.accountId;
     }
     headers['X-Snowth-Advisory-Limit'] = isLimited ? this.resultsLimit : "none";
+    headers['Accept'] = 'application/json';
     if (irondbOptions['std']['names'].length) {
+      var paneltype = irondbOptions['std']['names'][0]['leaf_data']['paneltype'] || "Graph";
+      options = {};
+      options.url = this.url;
+      if ('hosted' === this.irondbType) {
+        options.url = options.url + '/irondb';
+        options.url = options.url + '/fetch';
+      }
+      options.method = 'POST';
+      if ('standalone' === this.irondbType) {
+        options.url = options.url + '/fetch';
+      }
+      var metricLabels = [];
+      var interval = irondbOptions['std']['interval'];
+      var start = irondbOptions['std']['start'];
+      var end = irondbOptions['std']['end'];
+      start = Math.floor(start - (start % interval));
+      end = Math.floor(end - (end % interval));
```

Comment:
Are we concerned that the end truncate won't allow you to look at "now" until the next period interval?  If end % interval != 0, maybe we should += interval as well to make sure the desired end-time is inside the range?  e.g. "start" bumps itself backwards, and "end" bumps itself forward.

Reply:
if you bump it forward, then you would get a datapoint that is after the viewport ends.  that's a comment, not an opinion on what we should do.  I think it might be the right thing to do if grafana doesn't freak out.